app/shared/design-system/components/primitives/createDropdown.js

```diff
@@ -0,0 +1,151 @@
+/**
+ * @description
+ * - creates an agnostic dropdown menu
+ * - features a text field and select list
+ * - selecting a value from the select list updates the textfield
+ * - pressing "Escape" destroys the dropdown
+ * - clicking outside the dropdown container destroys the dropdown
+ * - using keyboard up/down keys navigates the list while focused
+ * - when navigating the list with arrow keys repeats from the start/end as per the last navigation point
+ * @example
+ * document.querySelector('button').addEventListener('pointerup', (event) => {
+ * 	 createDropdown(event, {
+ * 	   items: [
+ * 		   {value: 'px', selected: true},
+ * 		   {value: 'cm', selected: false},
+ * 		   {value: 'em', selected: false},
+ * 		   {value: 'rem', selected: false},
+ * 	   ],
+ *   });
+ * })
+ * @todo calculate if clientX + clientWidth is greater than viewport clientWidth and adjust as needed
+ * @todo stories file
+ * @todo api to recieve the resolved value
+ * @todo make the text field an optional feature, some dropdowns won't need it.
+ */
+class Dropdown extends HTMLElement {
+  connectedCallback() {
+    const impostor = document.querySelector(this.localName);
+    // there can only be one
+    if (impostor !== null && impostor !== this) impostor?.remove();
+    // immediately
+    this.dialogNode.show();
+    // focus select
+    this.selectNode.focus();
+  }
+  constructor() {
+    super();
+    this.innerHTML = `
+		<dialog>
+			<form method="dialog">
+				<fieldset>
+					<input autofocus placeholder="enter a value" value="${
+            this.querySelector("option[selected]")?.value || ""
```

Comment:
Idk, querying the thing ... 
it feels like we are back in time, first rendering templates then querying ...

Reply:
I feel like this should be a react component